test/spec/auth/connection-string.json

```diff
@@ -108,6 +143,16 @@
         }
       }
     },
+    {
+      "description": "must raise an error when the authSource is empty",
```

Comment:
This test and the one immediately below implement NODE-2443

Reply:
Linked in JIRA ticket.